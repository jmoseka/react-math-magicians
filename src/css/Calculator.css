.body {
  display: flex;
  justify-content: center;
  height: 100vh;
  font-size: 18px;
}

.container-calc {
  display: flex;
  justify-content: space-between;
}

.screen-output {
  background: #858694;
  padding: 15px;
  text-align: end;
  color: white;
}

button {
  margin: 0;
  padding: 15px;
  border: none;
  width: 100%;
  color: #1c1c24;
  border-right: 1px solid #b7b8bb;
  border-bottom: 1px solid #b7b8bb;
  background: #e0e0e0;
}

button:last-child {
  background: #f3933c;
}
