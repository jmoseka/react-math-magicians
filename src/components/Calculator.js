import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import React, { useState } from 'react';
import calculate from '../logic/calculate';
import '../css/Calculator.css';

function Calculator() {
  const [state, setState] = useState({});

  const btnClick = (e) => {
    setState(() => calculate(state, e.target.innerText));
  };
  const { next, total } = state;

  return (
    <div className="body">
      <div className="container col-10 col-md-4">
        <div className="row">
          <div className="col-12 screen-output">
            {next || total || 0}
          </div>
        </div>

        <div className="row">
          <button onClick={btnClick} type="button" className="col-3">AC</button>
          <button onClick={btnClick} type="button" className="col-3">/</button>
          <button onClick={btnClick} type="button" className="col-3">%</button>
          <button onClick={btnClick} type="button" className="col-3">+</button>
        </div>

        <div className="row">
          <button onClick={btnClick} type="button" className="col-3">7</button>
          <button onClick={btnClick} type="button" className="col-3">8</button>
          <button onClick={btnClick} type="button" className="col-3">9</button>
          <button onClick={btnClick} type="button" className="col-3">x</button>
        </div>

        <div className="row">
          <button onClick={btnClick} type="button" className="col-3">4</button>
          <button onClick={btnClick} type="button" className="col-3">5</button>
          <button onClick={btnClick} type="button" className="col-3">6</button>
          <button onClick={btnClick} type="button" className="col-3">-</button>
        </div>

        <div className="row">
          <button onClick={btnClick} type="button" className="col-3">1</button>
          <button onClick={btnClick} type="button" className="col-3">2</button>
          <button onClick={btnClick} type="button" className="col-3">3</button>
          <button onClick={btnClick} type="button" className="col-3">+</button>
        </div>

        <div className="row">
          <button onClick={btnClick} type="button" className="col-9">0</button>
          <button onClick={btnClick} type="button" className="col-3">=</button>
        </div>

      </div>
    </div>
  );
}

export default Calculator;
