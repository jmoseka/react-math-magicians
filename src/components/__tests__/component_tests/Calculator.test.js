import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line import/extensions
import Calculator from '../../calculator.js';

afterEach(() => {
  cleanup();
});

test('should match snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
