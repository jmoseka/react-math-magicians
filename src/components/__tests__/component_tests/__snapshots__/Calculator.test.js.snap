// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="body"
  data-testid="calc-1"
>
  <div
    className="container container-calc mt-5"
  >
    <div
      className="heading"
    >
      <h3>
        Let's do some Maths
      </h3>
    </div>
    <div
      className="col-6"
    >
      <div
        className="row"
      >
        <div
          className="col-12 screen-output"
        >
          0
        </div>
      </div>
      <div
        className="row"
      >
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          /
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="col-9"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="col-3"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
